src/Core/Client.vala

```diff
@@ -112,6 +102,10 @@ public class AppCenterCore.Client : Object {
     }
 
     private void updates_changed_callback () {
+        if (!has_tasks ()) {
+            update_restart_state ();
+        }
+
         var time_since_last_action = (new DateTime.now_local ()).difference (last_action) / GLib.TimeSpan.MILLISECOND;
         if (!has_tasks () && time_since_last_action >= PACKAGEKIT_ACTIVITY_TIMEOUT_MS) {
```

Comment:
I know I'm nitpicking now, but you could check `!has_tasks ()` only once in this function like this:

```
if (!has_tasks ()) {
    update_restart_state ();

    var time_since_last_action = (new DateTime.now_local ()).difference (last_action) / GLib.TimeSpan.MILLISECOND;
    if (time_since_last_action >= PACKAGEKIT_ACTIVITY_TIMEOUT_MS) {
       ...
   }
}
```